docs/source/python/compute.rst

```diff
@@ -292,4 +292,71 @@ The resulting dataset will be an :class:`.InMemoryDataset` containing the joined
    animal: [["Brittle stars",null,null]]
    n_legs: [[5,null,null]]
 
+.. _py-filter-expr:
 
+Filtering by Expressions
+========================
+
+:class:`.Table` and :class:`.Dataset` can
+both be filtered using a boolean :class:`.Expression`.
+
+The expression can be built starting from a 
+:func:`pyarrow.field`. Comparisons and transformations
+can then be applied to one or more fields to build the filter
+expression you care about.
+
+Most :ref:`compute` can be used to perform transformations
+on a ``field``.
+
+For example we could build a filter to find all rows that are even
+in column ``"nums"``
+
+.. code-block:: python
+
+   import pyarrow.compute as pc
+   even_filter = (pc.bit_wise_and(pc.field("nums"), pc.scalar(1)) == pc.scalar(0))
```

Comment:
I think this is a quite advanced example, as you need to understand what bitwise operations do on integers (which I don't think is common knowledge among data analysts, I also had to look it up). But unfortunately the remainder is not yet available (https://issues.apache.org/jira/browse/ARROW-12755?), otherwise could have done the equivalent of `(arr % 2) == 0`

Reply:
Uhm, I don't think is really advanced, I mean, it just checks the last bit of a number and I think it's well known to most people dealing with computers that the last bit of a number signals if it's a multiple of 2 or not.

I can add a `note` to explain what's going on so that we can make sure even people that don't remember that know what's going on.